### ISSUE TITLE
Unable to seal the key to nvram

### ISSUE THREAD

**Issue by stratoss** [fox-it/linux-luks-tpm-boot#2]
`seal-nvram.sh` and `seal-nvram.sh -z` both return this error after entering the owner password:

> Successfully created NVRAM area at index 0x1 (1).
> Tspi_NV_WriteValue failed: 0x00000001 - layer=tpm, code=0001 (1), Authentication failed

and `getsecret.sh` returns 16 FF's at index 1

**Comment by morbitzer** [comment]
So it seems that creating the NVRAM area with ``tpm_nvdefine`` works normally, but ``tpm_nvwrite`` fails. 

Did you use the well known secret for your SRK?

Do you maybe have a character in your password that could cause troubles in the bash-script (backslash, etc..)? Could you set the owner password to something easy (like "test") to see if this works?

What you could also try is to use another NVRAM area (set ``INDEX=<ara>`` in seal_nvram.sh). 

Could you manually try to write the keyfile into the NVRAM area and read it with 
``sudo tpm_nvdefine -i 2 -s 256 -p "AUTHWRITE" -z -o"<ownerpass>"`` 
``sudo tpm_nvwrite -i 2 -f /secret.bin -p"<ownerpass>" -z``
``sudo tpm_nvread -i 2``

**Comment by stratoss** [comment]
Yes, I'm using the well known secret for SRK. Will execute those tests tonight.

**Comment by Depaulicious** [comment]
I was having the same issue. Commenting the command out and running it manually in a shell worked around it.

**Comment by SuperSpyTX** [comment]
Move the -z before the -o parameter in seal-nvram in order for it to work. I'm guessing the -o parameter picks up on the -z.

Eg
```
tpm_nvwrite -i$INDEX -f $KEYFILE -z -p$OWNERPW
```

instead of
```
tpm_nvwrite -i$INDEX -f $KEYFILE -z -p$OWNERPW
```

Also you can use `READ_STCLEAR` permission so that after you have read from the NVRAM, you can actually just do `tpm_nvread -i $INDEX -s 0 -f /dev/null` so that the secret key can no longer be read until next boot.

**Comment by morbitzer** [comment]
I did change the position of the -z parameter. On my system, both versions work, so I can not test it. The new version is [here](https://github.com/morbitzer/linux-luks-tpm-boot). Let me know if it works. If it does, I will create a pull request. 

I've also added functionality to make use of `READ_STCLEAR`. Really nice idea! Initially, I thought about changing a PCR value after loading the secret to avoid reading it again, but this solution is easier. The only problem I faced is that for some reason the `tcsd` daemon stops after one `tpm_*` command is executed, so I have to restart the daemon in `getsecret.sh` and wait one second for it to start up.
